### ISSUE TITLE
"Undefined symbols for architecture arm64"

### ISSUE THREAD

**Issue by mtsibulski** [testfairy/testfairy-cordova-plugin#32]
My environment:
MacOS Big Sur 11.2.1 (20D74)
Xcode Version 12.4 (12D4e)

full error:
```
Undefined symbols for architecture arm64:
  "___llvm_profile_runtime", referenced from:
      ___llvm_profile_runtime_user in libTestFairy.a(TFFeedbackDrawView.o)
      ___llvm_profile_runtime_user in libTestFairy.a(TFDelegateImpl.o)
      ___llvm_profile_runtime_user in libTestFairy.a(TFMotionEndedClassVisitor.o)
      ___llvm_profile_runtime_user in libTestFairy.a(TFAsn1Parser.o)
      ___llvm_profile_runtime_user in libTestFairy.a(TFRSAUtils.o)
      ___llvm_profile_runtime_user in libTestFairy.a(TFPerformanceUtils.o)
      ___llvm_profile_runtime_user in libTestFairy.a(TFAESUtils.o)
      ...
     (maybe you meant: ___llvm_profile_runtime_user)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

i have tracked this down to be caused by the [latest 1.28.6 Spec published](https://github.com/CocoaPods/Specs/commits/master/Specs/b/0/e/TestFairy)

since the [plugin.xml](https://github.com/testfairy/testfairy-cordova-plugin/blob/master/plugin.xml#L25) has `spec="~> 1.28.2"`, it started using the broken spec. if you remove the `~>` to explicitly set the version, it works fine. i tested with spec version 1.28.5 and it was working ok for me, seems that 1.28.6 is the problematic one.

im not sure if we should explicitly set the spec version in this plugin.xml, or if there is an actual issue with the podspec, or maybe both. im not sure how to report an issue abotu the podspec, so im reporting it here

fyi, you may also want to consider moving away from using the `<framework>` tag and instead use `<podspec>`, as im getting this warning from cordova about it: **"framework" tag with type "podspec" is deprecated and will be removed. Please use the "podspec" tag**. just note that this would also force your users to use cordova-ios@5 and cordova@9

example working diff:
```
diff --git a/testfairy-cordova-plugin/plugin.xml b/testfairy-cordova-plugin/plugin.xml
index 60856846dc0..55b2ac70083 100755
--- a/testfairy-cordova-plugin/plugin.xml
+++ b/testfairy-cordova-plugin/plugin.xml
@@ -22,7 +22,14 @@
         </config-file>
         <header-file src="src/ios/CDVTestFairy.h" />
         <source-file src="src/ios/CDVTestFairy.m" />
-        <framework src="TestFairy" type="podspec" spec="~> 1.28.2" />
+        <podspec>
+            <config>
+                <source url="https://cdn.cocoapods.org/"/>
+            </config>
+            <pods use-frameworks="true">
+                <pod name="TestFairy" spec="1.28.5" />
+            </pods>
+        </podspec>
     </platform>
     <platform name="android">
         <source-file src="src/android/CDVTestFairy.java" target-dir="src/com/testfairy" />
```

**Comment by vijaysharm** [comment]
Hi @mtsibulski thanks for reporting this issue. I haven't seen this issue before, so I'm very interested in reproducing the issue. Does this happen when you build an iOS project? We have some steps that build a sample app in the `.travis.yml`. Does the issue happen there? I'll try it on my end using the same version of the spec to see if i get the same issue, but if you have some steps i could follow to reproduce the issue i'd appreciate it. 

Also thanks for identifying that we're not defining the specific version of the SDK. We prefer to bind the version of the corodova library with the specific version of the android and ios sdk for support reasons. Thank you for sending me a patch, if you want to submit a PR, I'm happy to merge it into our repo, if not, I'll be adding the patch after trying to reproduce the above issue.

**Comment by jamesyoon11** [comment]
@vijaysharm  I can reproduce the same error, too.

Reproduce steps:
```
pod repo update
npx cordova create SampleApp com.sample.app SampleApp
cd SampleApp/
npx cordova platform add ios
npx cordova plugin add com.testfairy.cordova-plugin
npx cordova build ios --simulator
```

It will throw
```
Undefined symbols for architecture x86_64:
  "___llvm_profile_runtime", referenced from:
      ___llvm_profile_runtime_user in libTestFairy.a(TFFeedbackDrawView.o)
      ___llvm_profile_runtime_user in libTestFairy.a(TFDelegateImpl.o)
      ___llvm_profile_runtime_user in libTestFairy.a(TFMotionEndedClassVisitor.o)
      ___llvm_profile_runtime_user in libTestFairy.a(TFAsn1Parser.o)
      ___llvm_profile_runtime_user in libTestFairy.a(TFRSAUtils.o)
      ___llvm_profile_runtime_user in libTestFairy.a(TFPerformanceUtils.o)
      ___llvm_profile_runtime_user in libTestFairy.a(TFAESUtils.o)
      ...
     (maybe you meant: ___llvm_profile_runtime_user)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I compared `libTestFairy.a` between 1.28.5 and 1.28.6.

`1.28.5` does not return undefined symbol `___llvm_profile_runtime`.

```
TestFairySDK-1.28.5$ nm -Au libTestFairy.a | grep ___llvm_profile_runtime
TestFairySDK-1.28.5$
```

However, `1.28.6` returns undefiend sysmbol  `___llvm_profile_runtime`.
```
TestFairySDK-1.28.6$ nm -Au libTestFairy.a | grep ___llvm_profile_runtime
libTestFairy.a:TFFeedbackDrawView.o: ___llvm_profile_runtime
libTestFairy.a:TFNewRelicFeature.o: ___llvm_profile_runtime
libTestFairy.a:TFDelegateImpl.o: ___llvm_profile_runtime
libTestFairy.a:TFInstalledFrameworkClassVisitor.o: ___llvm_profile_runtime
libTestFairy.a:TFMotionEndedClassVisitor.o: ___llvm_profile_runtime
libTestFairy.a:TFAsn1Parser.o: ___llvm_profile_runtime
...
TestFairySDK-1.28.6$
```

So, I think the 1.28.6 library was built with code coverage enabled.

https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/testing_with_xcode/chapters/07-code_coverage.html

If I turn on the code coverage in the project, the error does not happen. However, the cordova build does not turn on code coverage by default, `libTestFairy.a` should be fixed.

**code coverage disabled:**
<img width="1193" alt="code coverage disabled" src="https://user-images.githubusercontent.com/13875691/116322784-9baecf80-a771-11eb-875b-750370d632d9.png">


**code coverage enabled:**
<img width="1157" alt="code coverage enabled" src="https://user-images.githubusercontent.com/13875691/116322817-af5a3600-a771-11eb-9959-a6cd64de0799.png">

**Comment by vijaysharm** [comment]
That really awesome @mtsibulski I appreciate your time with this. I've made this my top priority and I'll investigate what's going on. We recently moved our minimum support from Xcode 11 to Xcode 12, and this may be an issue from that change. Again I appreciate your help here.

**Comment by vijaysharm** [comment]
Hi @mtsibulski It took a while to track down, but I believe we've resolved the issue in libTestFairy.a. I've released a version of both the native sdk and the cordova plugin. The fix should be included in version [2.48.0](https://github.com/testfairy/testfairy-cordova-plugin/releases/tag/2.48.0). Could you give it a try and let me know if it works for you.

Also, I was wondering if you'd like to contribute your patch for how we reference the podspec in this project? If not I'll work on updating that today.

I want to thank you again for your help in identifying this issue. I sincerely appreciate your thorough report of the issue. 🍻

**Comment by mtsibulski** [comment]
@vijaysharm Thanks for the quick turnaround! I gave it a quick test and it seems to be working now. I verified using the test scenario provided by my colleague @jamesyoon11 , as well as against our local applications.

I will be submitting a PR shortly with the new podspec reference. Btw I just realized that not having an explicitly defined podspec on older versions of this plugin is now working in our favor, since users will end up skipping over the broken version haha.

And you are very welcome; and thank you as well! James and I are very pleased with the rapid response and fix. You don't see that too often on cordova projects, unfortunately.
